test/jdk/java/awt/font/FontScaling/RotatedScaledFontTest.java

```diff
@@ -128,4 +137,14 @@ private static Rectangle findTextBoundingBox(BufferedImage image) {
     private static boolean roughlyEqual(int x1, int x2, int scale) {
         return Math.abs(x1 - x2) <= Math.ceil(scale / 2d) + 1; // higher scale = higher allowed variance
     }
+
+    private static String saveImage(String name, BufferedImage image) {
```

Comment:
is this a good idea to write files?
e.g. the CI system may not make the files accessible, the local FS might overflow, etc.

there is an idea to output base-64 encoded image to the stdout, see https://bugs.openjdk.org/browse/JDK-8328716
this way
- the images will be available as long as the log file is available
- zero dependency on local file system or its limitations

Reply:
> is this a good idea to write files? e.g. the CI system may not make the files accessible, the local FS might overflow, etc.
> 
> there is an idea to output base-64 encoded image to the stdout, see https://bugs.openjdk.org/browse/JDK-8328716 this way
> 
> * the images will be available as long as the log file is available
> * zero dependency on local file system or its limitations

It is perfectly fine. We do this in many tests. jtreg ensures that CWD is the working directory of the test.
it will thus be preserved along with the rest of the files jtreg preserves.
And saving as .png lets you look at it without more hassle.
OTOH I don't think it necessary to print the absolute path.
Often that's on a test system you don't get to see. Instead you just get to downlaod the results and the absolute path is not a good idea then.

Reply:
Interesting idea. I personally don't know enough to weigh in on whether the benefits of saving to disk outweigh the drawbacks. It might be possible to emit the Base64-ed PNG to sysout or syserr (these images should compress very well as PNG), but keep in mind that these two tests are headless and drawing to an in-memory `BufferedImage`, so the usage might be a little different than the screenshot-focused solution outlined in JDK-8328716.

Reply:
> Interesting idea. I personally don't know enough to weigh in on whether the benefits of saving to disk outweigh the drawbacks. It might be possible to emit the Base64-ed PNG to sysout or syserr (these images should compress very well as PNG), but keep in mind that these two tests are headless and drawing to an in-memory `BufferedImage`, so the usage might be a little different than the screenshot-focused solution outlined in JDK-8328716.

There is no issue saving to disk.
If there were, we'd have big problems because that's where the class files go !
Doing what I suggested is the way to go.

Reply:
> It is perfectly fine. We do this in many tests.

OK, thanks for the clarification.

> OTOH I don't think it necessary to print the absolute path.

> Doing what I suggested is the way to go.

Just to be clear then -- the test should continue to save the images to disk on failure, but should not include the file path in the error message, correct? I can update the PR if that's the case.

Reply:
Right. 
And actually the best thing when saving the file would be to to get the dir path to save to from using
something like this
String dir = System.getProperty("test.classes",  ".");
String filename = dir + java.io.File.separator + basename + ".png"; 
This ensures it is saved the same place as jtreg puts the class files and if that is not defined (ie you are running it outside jtreg), then in the current dir.
But no need to "print" that path.

Reply:
Great, I've updated the PR accordingly.